internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1634,6 +1634,10 @@ func newAssistedServiceDeployment(ctx context.Context, log logrus.FieldLogger, a
 			newSecretEnvVar("POSTGRESQL_DATABASE", "db.name", databaseName),
 			newSecretEnvVar("POSTGRESQL_USER", "db.user", databaseName),
 			newSecretEnvVar("POSTGRESQL_PASSWORD", "db.password", databaseName),
+			// allow auto upgrade when new image is available
+			// allows upgrade for +1 version only, and cannot skip versions
+			newSecretEnvVar("POSTGRESQL_UPGRADE", "POSTGRESQL_UPGRADE", "true"),
```

Comment:
most likely not from secret

Reply:
Right, you should just define the `corev1.EnvVar` directly here.